invenio_sitemap/tasks.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+# Copyright (C) 2025 Northwestern University.
+#
+# invenio-sitemap is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Tasks."""
+
+from datetime import datetime, timezone
+
+from celery import shared_task
+from flask import current_app
+from invenio_base import invenio_url_for
+from invenio_cache import current_cache
+
+from .cache import SitemapCache, SitemapIndexCache
+from .utils import format_to_w3c, parse_from_w3c
+
+
+@shared_task(ignore_results=True)
+def update_sitemap_cache():
+    """Update the cache with sitemap related content."""
+    # Sitemaps
+    max_count = current_app.config["SITEMAP_MAX_ENTRY_COUNT"]
+    lastmods = []
+    cache_sitemap = SitemapCache(current_cache)
+    i = 0
+    for batch in batched(_iterate_url_entries(), max_count):
+        cache_sitemap.set(i, batch)
+        lastmods.append(_get_latest_lastmod(batch))
+        i += 1
+
+    # Delete any cached entry that are now stale
+    # Usually, this would delete at most 1 entry since there is rarely a mass
+    # removal between runs. This is more efficient than previous indiscriminate clearing
+    cache_sitemap.delete_included_and_higher(i)
+
+    # Sitemap indices
+    cache_sitemap_index = SitemapIndexCache(current_cache)
+    j = 0
+    for batch in batched(_iterate_sitemap_entries(lastmods), max_count):
+        cache_sitemap_index.set(j, batch)
+        j += 1
+
+    # Delete any cached entry that are now stale
+    # again, unlikely that mass deletions occur
+    cache_sitemap_index.delete_included_and_higher(j)
+
+
+def batched(iterable, n):
```

Comment:
If we add this primarily for <3.12 compatibility, should we directly copy the function as described on https://docs.python.org/3/library/itertools.html#itertools.batched ?

Reply:
I was thinking the walrus operator was 3.10 but it's 3.8 (argh, time flies!). But I don't care for it too much and since we don't use the strict option, it's probably as-is since it's going away eventually.